executors/cmdexec/block_volume.go

```diff
@@ -293,6 +294,26 @@ func (c *CmdExecutor) BlockVolumeInfo(host string, blockhostingvolume string,
 	blockVolumeInfo.Username = blockVolumeInfoExec.Gbid
 	blockVolumeInfo.Password = blockVolumeInfoExec.Password
 
+	// From gluster-block output,
+	// get the Minimum Size from the list of Resize failed nodes
+	var differSizes []int
+	for _, size := range blockVolumeInfoExec.ResizeFailed {
```

Comment:
Cant you do this in one loop? Pseudo-code:
```
minSize := -1
for _, size := range blockVolumeInfoExec.ResizeFailed {
    value, err := toGigaBytes(size)
    if err != nil {
        return err
    }
    if minSize == -1 || value < minSize {
        value = size {
    }
}
if minSIze != -1 {
    blockVolumeInfo.UsableSize = minSize
}
```

Also, please don't silently drop the error from toGigaBytes by just passing over it in the loop. If that function fails in production the best thing heketi can do is fail the operation noisily.

Reply:
ok